### PR TITLE
refactor(GCS+gRPC): helpers for bucket names

### PR DESCRIPTION
**Pull request by coryan**
Motivated by #5673

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9788)

### CODE REVIEW

File: google/cloud/storage/internal/grpc_bucket_name.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+std::string BucketNameToProto(std::string const& bucket_name);
+std::string BucketNameFromProto(std::string const& bucket_name);
```
Comment (devbww) [inline google/cloud/storage/internal/grpc_bucket_name.h:28]: I'm not sure "proto" enters into this.  Would these be better as something like `BucketNameToId(bucket_name)` and `BucketIdToName(bucket_id)`?

**Review by dbolduc — APPROVED** [review]
* you could add unit tests
* you could add a `NotifcationNameToProto(std::string const& bucket_name, std::string const& notification_id)`

File: google/cloud/storage/internal/grpc_bucket_name.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/grpc_bucket_name.h]: I think the name should hint that it is gRPC related, as the `id` and `name` are identical for REST.  Please let me know if the new names make sense.

**Comment by coryan** [comment]
> * you could add unit tests

Done.

> * you could add a `NotifcationNameToProto(std::string const& bucket_name, std::string const& notification_id)`

Seems like not enough value for money.

**Review by devbww — APPROVED** [review]

**Pull request merged**